lib/express_validator.js

```diff
@@ -259,13 +263,32 @@ var expressValidator = function(options) {
  */
 
 function validateSchema(schema, req, loc, options) {
+  var locations = ['body', 'params', 'query'],
+    currentLoc = loc;
+
   for (var param in schema) {
-    loc = loc === 'any' ? locate(req, param) : loc;
-    var validator = new ValidatorChain(param, null, req, loc, options);
+
+    // check if schema has defined location
+    if (schema[param].hasOwnProperty('in')) {
+      if (locations.indexOf(schema[param].in) !== -1) {
+        currentLoc = schema[param].in;
+      } else {
+        // skip params where defined location is not supported
+        continue;
+      }
+    }
+
+    currentLoc = currentLoc === 'any' ? locate(req, param) : currentLoc;
+    var validator = new ValidatorChain(param, null, req, currentLoc, options);
     var paramErrorMessage = schema[param].errorMessage;
     delete schema[param].errorMessage;
 
     for (var methodName in schema[param]) {
+      // skip method if this is location definition
+      if (methodName === 'in') {
+        currentLoc = loc;
```

Comment:
In line 274 we are defining location per one parameter in whole schema, and this is only in case when `in` is defined. So for rest parameters we should use old, original location e.g.: user is calling `req.checkBody(schema);` but one of the parameters in schema has `in: 'query'`. Without reassigning original location after reading param with `in` rest of params (this without `in`), will be checked against query as well, but should be checked against body :)